File: src/main/java/voidchess/ui/TableInterface.java
package voidchess.ui;

import voidchess.board.MoveResult;
import voidchess.helper.Move;
import voidchess.player.PlayerInterface;

/**
 * @author stephan
 */
public interface TableInterface {
    void startGame();
    void stopGame(MoveResult endoption);
    void move(Move move);
    void setWhitePlayer(PlayerInterface player);
    void setBlackPlayer(PlayerInterface player);
}

File: src/main/kotlin/voidchess/ui/TableInterface.kt
package voidchess.ui

import voidchess.board.move.MoveResult
import voidchess.board.move.Move
import voidchess.player.PlayerInterface


interface TableInterface {
    fun startGame()
    fun stopGame(endoption: MoveResult)
    fun move(move: Move)
    fun setWhitePlayer(player: PlayerInterface)
    fun setBlackPlayer(player: PlayerInterface)
    fun switchPlayer()
}
